sdk/internal/perf/atomic_float.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package perf
+
+import (
+	"sync"
+)
+
+type atomicFloat64 struct {
```

Comment:
Something to consider in a future PR - would it be simpler to store off the time.Duration values, which fit into an int64 and just use the plain `atomic` functions instead? 

(I only briefly looked but it looks like this is meant to be "fractional seconds", but you really can only measure to the granularity of the time keeping in Go anyways)

Reply:
This worked out fine, there's a couple extra type conversions that are a bit annoying but it's cleaner than rolling a new atomic type.